qiskit/compiler/transpile.py

```diff
@@ -201,28 +207,72 @@ def callback_func(**kwargs):
                                   'is greater than maximum ({}) '.format(max_qubits) +
                                   'in the coupling_map')
     # Transpile circuits in parallel
-    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_configs)))
+    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_args)))
 
     if len(circuits) == 1:
         return circuits[0]
     return circuits
 
 
-# FIXME: This is a helper function because of parallel tools.
 def _transpile_circuit(circuit_config_tuple):
     """Select a PassManager and run a single circuit through it.
-
     Args:
         circuit_config_tuple (tuple):
             circuit (QuantumCircuit): circuit to transpile
-            transpile_config (TranspileConfig): configuration dictating how to transpile
-
+            transpile_config (dict): configuration dictating how to transpile. The
+                dictionary has the following format:
+                {'optimization_level': int,
+                 'pass_manager': PassManager,
+                 'output_name': string,
+                 'callback': callable,
+                 'pass_manager_config': PassManagerConfig}
     Returns:
         QuantumCircuit: transpiled circuit
+    Raises:
+        TranspilerError: if transpile_config is not valid or transpilation incurs error
     """
     circuit, transpile_config = circuit_config_tuple
 
-    return transpile_circuit(circuit, transpile_config)
+    pass_manager_config = transpile_config['pass_manager_config']
+
+    # Workaround for ion trap support: If basis gates includes
+    # Mølmer-Sørensen (rxx) and the circuit includes gates outside the basis,
+    # first unroll to u3, cx, then run MSBasisDecomposer to target basis.
+    basic_insts = ['measure', 'reset', 'barrier', 'snapshot']
+    device_insts = set(pass_manager_config.basis_gates).union(basic_insts)
+    ms_basis_swap = None
+    if 'rxx' in pass_manager_config.basis_gates and \
+            not device_insts >= circuit.count_ops().keys():
+        ms_basis_swap = pass_manager_config.basis_gates
+        pass_manager_config.basis_gates = list(
+            set(['u3', 'cx']).union(pass_manager_config.basis_gates))
+
+    if transpile_config['pass_manager'] is not None:
```

Comment:
there were some code comments here that got lost during copy/paste. comments are good :)

Reply:
commnets in b84e270ac83a1dbfc3d493da7abe69751843f8bd